platform-api/src/main/java/io/datacater/core/kubernetes/PythonRunnerPool.java

```diff
@@ -116,206 +94,100 @@ private String prependSlash(String path) {
     };
   }
 
-  Function<AsyncMap<String, RunnerPool>, Uni<RunnerPool>> fillQueue =
-      map -> {
-        List<Pod> pods =
-            kubernetesClient
-                .pods()
-                .inNamespace(DataCaterK8sConfig.NAMESPACE)
-                .withLabels(DataCaterK8sConfig.LABELS)
-                .list()
-                .getItems();
-        Deque<NamedPod> namedPods =
-            new ArrayDeque<NamedPod>(DataCaterK8sConfig.PYTHON_RUNNER_REPLICAS);
-        for (Pod pod : pods) {
-          String podName = pod.getMetadata().getName();
-          namedPods.add(new NamedPod(podName, pod));
-        }
-
-        return map.get(POOL_NAME)
-            .onItem()
-            .ifNull()
-            .continueWith(new RunnerPool(namedPods))
-            .invoke(pool -> map.put(POOL_NAME, pool))
-            .onItem()
-            .ifNotNull()
-            .invoke(pool -> map.replace(POOL_NAME, pool));
-      };
-
-  Function<Lock, Pod> lockTransformerUni =
-      lock -> {
-        Uni<AsyncMap<String, Pod>> pods = sharedData.getAsyncMap(POOL_NAME);
-        lock.release();
-        return pods.onItem().transform(locki -> new Pod()).await().indefinitely();
-      };
-
-  Function<AsyncMap<String, Deque<NamedPod>>, NamedPod> transformAsyncMapToPod =
-      map ->
-          map.get(POOL_NAME)
-              .onItem()
-              .transform(Deque::pop)
-              .onFailure()
-              .transform(
-                  Unchecked.function(
-                      throwable -> {
-                        throw new DatacaterException(throwable.getMessage());
-                      }))
-              .await()
-              .indefinitely();
-
-  UnaryOperator<Pod> annotatePodAsUsed =
-      pod -> {
-        Map<String, String> annotations = pod.getMetadata().getAnnotations();
-        String podName = pod.getMetadata().getName();
-
-        if (annotations.containsKey(IN_USE_LABEL_KEY)
-            && annotations.get(IN_USE_LABEL_KEY).equals(IN_USE_TRUE_LABEL_VALUE)) {
-          throw new DatacaterException(
-              String.format(
-                  "Pod with Name := %s is annotated as in use. Will not re-annotate.", podName));
-        }
-
-        return kubernetesClient
-            .pods()
-            .inNamespace(DataCaterK8sConfig.NAMESPACE)
-            .withName(podName)
-            .edit(
-                p ->
-                    new PodBuilder(p)
-                        .editOrNewMetadata()
-                        .addToAnnotations(IN_USE_TRUE_LABEL)
-                        .endMetadata()
-                        .build());
-      };
-
-  private Boolean isStatefulSetActive() {
-    int statefulSetCount =
+  private Deque<NamedPod> podsFromKubernetes() {
+    List<Pod> pods =
         kubernetesClient
-            .apps()
-            .statefulSets()
+            .pods()
             .inNamespace(DataCaterK8sConfig.NAMESPACE)
             .withLabels(DataCaterK8sConfig.LABELS)
```

Comment:
Would a request to a pod, which is in the `Terminating` or `Waiting` state, instantly fail or wait until the pod is `Running`?

If it would immediately fail, it could make sense to only select pods that are in a [Running](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-state-running) state.